lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,6 +71,7 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
+          return serializer._type unless serializer._type.nil?
```

Comment:
How about using `.present?` here? As long as we rely on `ActiveSupport` and empty type field doesn't make too much sense... Using variable directly in conditional would work, but I'm trying to avoid such things. Just in case. :)